c7n/resources/licensemanager.py

```diff
@@ -0,0 +1,259 @@
+
+    class resource_type(object):
+        service = 'license-manager'
+        enum_spec = ('list_license_configurations', 'LicenseConfigurations', None)
+        detail_spec = (
+            'get_license_configuration', 'LicenseConfigurationArn',
+            'LicenseConfigurationArn', None)
+        id = 'LicenseConfigurationArn'
+        name = 'Name'
+        date = None
+        dimension = None
+        filter_name = None
+
+    permissions = ('license-manager:ListTagsForResource',)
+
+    def augment(self, resources):
+        client = local_session(self.session_factory).client('license-manager')
+
+        def _augment(r):
+            # List tags for the license_configuration & set as attribute
+            tags = self.retry(client.list_tags_for_resource,
+                ResourceArn=r['LicenseConfigurationArn'])['Tags']
+            r['Tags'] = tags
+            return r
+
+        # Describe license_configuration & then list tags
+        resources = super(LicenseConfiguration, self).augment(resources)
+        with self.executor_factory(max_workers=1) as w:
+            return list(filter(None, w.map(_augment, resources)))
+
+
+class StateTransitionFilter(object):
+    """Filter instances by state.
+
+    Try to simplify construction for policy authors by automatically
+    filtering elements (filters or actions) to the instances states
+    they are valid for.
+
+    """
+    valid_origin_states = ()
+
+    def filter_instance_state(self, instances, states=None):
+        states = states or self.valid_origin_states
+        orig_length = len(instances)
+        results = [i for i in instances
+                   if i['LicenseDeletionStatus'] in states]
+        self.log.info("state filter %s %d of %d license deletion" % (
+            self.__class__.__name__, len(results), orig_length))
+        return results
+
+
+@LicenseConfiguration.action_registry.register('delete')
+class DeleteLicenseConfiguration(BaseAction, StateTransitionFilter):
+    """Deletes LicenseConfiguration(s)
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: delete-license-Configuration
+            resource: license-configuration
+            filters:
+              - "tag:DeleteMe": present
+            actions:
+              - delete
+    """
+    schema = type_schema('delete')
+    permissions = ('license-manager:DeleteLicenseConfiguration',)
+
+    def process(self, resources):
+        if not len(resources):
+            return
+
+        client = local_session(self.manager.session_factory).client('license-manager')
+
+        for n in resources:
+            try:
+                client.delete_license_configuration(
+                    LicenseConfigurationArn=n['LicenseConfigurationArn'])
+            except client.exceptions.ResourceNotFound:
+                pass
+
+
+@resources.register('service-settings')
+class ServiceSettings(QueryResourceManager):
+
+    class resource_type(object):
```

Comment:
I have question regarding this service-settings method. while registering service-setting as a resource it complaining about enum_spec. I tried running test policy but we are not getting any resources back.i think that is because we do not have any id. Can you please point out me in the right direction?

Reply:
this shouldn't be a resource, it should be an account filter, please remove and submit as a separate pull request